builtin/credential/approle/path_role.go

```diff
@@ -550,22 +550,27 @@ func (b *backend) pathRoleSecretIDList(req *logical.Request, data *framework.Fie
 		// corresponding lock many times using SecretIDs is not
 		// possible. Also, indexing it everywhere using secretIDHMACs
 		// makes listing operation easier.
-		lock := b.secretIDLock(secretIDHMAC)
-		lock.RLock()
+		secretIDLock := b.secretIDLock(secretIDHMAC)
```

Comment:
It seems like the overall effect of this PR is to account for the case where secretIDHMAC == "" in which case maybe there's a better check that can be in place? (Or, the lock doesn't need to be locked earlier?)